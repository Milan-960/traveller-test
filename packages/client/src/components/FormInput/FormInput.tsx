import { Search2Icon } from '@chakra-ui/icons'
import { IconButton, Input, InputGroup, InputRightElement } from '@chakra-ui/react'
import React from 'react'

type FormProps = {
  onSubmit?: (e: any) => void
  initialValues?: string
}

export const FormInput: React.FC<FormProps> = ({ onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <InputGroup>
        <Input
          autoFocus
          placeholder="search your favorite country!!"
          type="text"
          _placeholder={{ color: 'blue.300' }}
          borderColor="blue.300"
          data-testid="inputAddress"
        />
        <InputRightElement>
          <IconButton aria-label="" data-testid="submitButton" type="submit" icon={<Search2Icon />} />
        </InputRightElement>
      </InputGroup>
    </form>
  )
}
