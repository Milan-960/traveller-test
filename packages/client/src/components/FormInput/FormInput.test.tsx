import { fireEvent, screen } from '@testing-library/react'
import React from 'react'
import { render } from '../../test-utils'
import FormInput from './FormInput'

describe('FormInput', () => {
  it('snapshot', () => {
    const wrapper = render(<FormInput onSubmit={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('change input', async () => {
    const wrapper = render(<FormInput onSubmit={jest.fn()} />)

    const inputChange = screen.getByTestId('inputAddress')
    //fireEvent.change isnt working correctly, can be targeted but innerHTML doesnt change
    fireEvent.change(inputChange, { target: { value: 'test' } })
    await new Promise(resolve => setTimeout(resolve, 100))
    expect(wrapper.findByText('test')).toBeInTheDocument()
  })

  it('click button form submit', async () => {
    const mockSubmit = jest.fn()
    const wrapper = render(<FormInput onSubmit={mockSubmit} />)

    const buttonClick = screen.getByTestId('submitButton')
    fireEvent.click(buttonClick)

    expect(mockSubmit).toHaveBeenCalled()
  })
})
