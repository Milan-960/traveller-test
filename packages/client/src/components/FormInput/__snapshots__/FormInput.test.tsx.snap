// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <form>
        <div
          class="chakra-input__group css-4302v8"
        >
          <input
            class="chakra-input css-1n546ci"
            data-testid="inputAddress"
            placeholder="search your favorite country!!"
            type="text"
          />
          <div
            class="chakra-input__right-element css-ur97uj"
          >
            <button
              aria-label=""
              class="chakra-button css-dn5esd"
              data-testid="submitButton"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="chakra-input__group css-4302v8"
      >
        <input
          class="chakra-input css-1n546ci"
          data-testid="inputAddress"
          placeholder="search your favorite country!!"
          type="text"
        />
        <div
          class="chakra-input__right-element css-ur97uj"
        >
          <button
            aria-label=""
            class="chakra-button css-dn5esd"
            data-testid="submitButton"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
